sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -530,9 +544,13 @@ public ServiceBusSenderAsyncClient buildAsyncClient() {
             if (!CoreUtils.isNullOrEmpty(viaQueueName) && entityType == MessagingEntityType.SUBSCRIPTION) {
                 throw logger.logExceptionAsError(new IllegalStateException(String.format(
                     "(%s), Via queue feature work only with a queue.", viaQueueName)));
+            } else if (!CoreUtils.isNullOrEmpty(viaTopicName) && entityType == MessagingEntityType.QUEUE) {
+                throw logger.logExceptionAsError(new IllegalStateException(String.format(
+                    "(%s), Via topic feature work only with a topic.", viaTopicName)));
             }
```

Comment:
What happens if both viaQueueName and viaTopicName are set.

Reply:
`validateEntityPaths` rules out that both topicName and queueName are provided.
If both `viaQueueName` and `viaTopicName` are set, one of the above two validations will throw an error.